### DISCUSSION TITLE
Appium 2.0.-beta20 : start server with plugin changed ?

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#16103] [Q&A]
Don't know if it is a bug, but i upgraded appium from 2.0.0-beta18 to beta21 and know i get an error on starting the server

`appium: error: unrecognized arguments: --plugins images --appium-home /Users/stephan.bast/.appium`

i create the server programmatically
`        builder = new AppiumServiceBuilder().withLogFile(new File("appium_"+className+".log")).withArgument(() -> "--appium-home", getAppiumHome()).withArgument(() -> "--base-path", "/wd/hub").withArgument(() -> "--plugins", "images");
`

Anyone an idea it there was a known changed here ?

**Comment by KazuCocoa** [comment]
Could you change `--appium-home` to `APPIUM_HOME` as an environment variable?
Both worked before, but we dropped the `--appium-home`.

**Reply by RSM-SBA** [reply]
thanks for the quick response, i will give it a try.
btw. is node@14 now recommend ? 
i have seen a warning
`npm WARN notsup Unsupported engine for klaw@4.0.1: wanted: {"node":">=14.14.0"} (current: {"node":"12.19.1","npm":"6.14.8"})
npm WARN notsup Not compatible with your version of node/npm: klaw@4.0.1`

**Reply by KazuCocoa** [reply]
no, we keep v12 for now.
the klaw library should keel lower one for us.
https://github.com/appium/appium/commit/6095fae8c365e962dac9904c3776bea53801e3a6#diff-f3e2ea0832e55ebd21e8648a60359be9b758c1776ae531b64119510c203399c4 . Let me revert it.

https://github.com/appium/appium/pull/16105

**Reply by RSM-SBA** [reply]
ok i have removed the argument from startup to APPIUM_HOME Enviroment variable, but now
i get 
`appium: error: unrecognized arguments: --plugins images
`

`        builder = new AppiumServiceBuilder().withLogFile(new File("appium_"+className+".log")).withArgument(() -> "--base-path", "/wd/hub").withArgument(() -> "--plugins", "images");
`

btw is there a good reason to remove it ?
Because for me it makes more complex to set the correct HOME Enviroment on different CI machines

**Reply by KazuCocoa** [reply]
I usually set it as a process environment variable like `APPIUM_HOME="/path/to/place" appium --arguments etc`

> btw is there a good reason to remove it ?

sorry, do you mean why we removed `--appium-home` argument?
APPIUM_HOME also existed at initial time to define appium-home. The main reason is to apply the appium-home to all plugins/drivers without additional implementation to parse arguments correctly. It helped to simplify appium's implementation.

**Reply by KazuCocoa** [reply]
`--plugins` probably it got a new format. @jlipps may know of it

**Reply by jlipps** [reply]
Yep, it's now `--use-plugins`

**Reply by jlipps** [reply]
(I have open PRs everywhere I can think of now to make sure `--use-plugins` is what is mentioned)

**Reply by RSM-SBA** [reply]
ok thanks, the argument --use-plugins works
But now there is another one :D

I got the error 
`An unknown server-side error occurred while processing the command. Original error: Potentially insecure feature 'adb_shell' has not been enabled`

But i have enabled relaxed security via AppiumServerBuilder Argument. Is this also removed ?


` builder.usingPort(4726);

		builder.withCapabilities(cap);
		
		builder.withArgument(GeneralServerFlag.SESSION_OVERRIDE);
		
		builder.withArgument(GeneralServerFlag.LOG_LEVEL, getAppiumLogLevel());
		
		builder.withArgument(GeneralServerFlag.RELAXED_SECURITY);
		
		builder.withArgument(GeneralServerFlag.LOG_NO_COLORS);
		
		builder.withArgument(GeneralServerFlag.CALLBACK_PORT, getAppiumPort());
		
		builder.withArgument(GeneralServerFlag.LOG_TIMESTAMP);
		
		builder.withArgument(GeneralServerFlag.LOCAL_TIMEZONE);`		

Are there also other arguments that have changed ? Is there a list anywhere what is removed/changed ?

**Reply by jlipps** [reply]
i'm not sure about builder. i think `--relaxed-security` has been gone for some time, in favour of e.g. `--allow-insecure=adb_shell` (where the specific features are listed in a comma separated way)

**Reply by RSM-SBA** [reply]
Hmm ok, the builder with relaxes security has worked in beta.18.
but the other way also works, thank you :)

**Reply by jlipps** [reply]
yeah, beta 20/21 released a totally new configuration system. @boneskull did we intentionally get rid of `--relaxed-security` somehow? and @StephanB75 no I don't know anything about the java client builder. is there no way to add arbitrary flags?

**Reply by boneskull** [reply]
@jlipps `--relaxed-security` was not removed... though it may not actually _do_ anything.  Is it basically an alias for `--allow-insecure=adb_shell`?

**Reply by jlipps** [reply]
it is theoretically an alias for `--allow-insecure=*` (if allow insecure allowed an 'all' code). i wouldn't have been surprised that it didn't work at all in appium 2. but the fact that it seemed to work in beta 18 of appium 2 and now doesn't makes me think of config changes as a possible culprit.

**Reply by boneskull** [reply]
hmm.  OK.  I'll come up w/ something.

**Reply by boneskull** [reply]
looks like the `relaxed-security` value should be stored in a prop named `relaxedSecurityEnabled`, which I missed.  it's a one-liner

**Reply by boneskull** [reply]
See https://github.com/appium/appium/pull/16117

**Reply by RSM-SBA** [reply]
@jlipps There is a way to add the flags and it works when adding 
`.withArgument(() -> "--allow-insecure", "adb_shell")`

Thank you all for your quick responses!
